pkg/controller/support/credentials/cloudauth/acr/acr.go

```diff
@@ -79,30 +78,33 @@ func newProvider(ctx context.Context, logger logr.Logger) (*acrProvider, error)
 	}
 
 	return &acrProvider{
-		logger:                logger.WithName("acr-auth-provider"),
 		tenantID:              settings.Values[auth.TenantID],
 		servicePrincipalToken: token,
 	}, nil
 }
 
-func (a *acrProvider) authenticate(ctx context.Context, server string) (*types.AuthConfig, error) {
+func (a *acrProvider) authenticate(ctx context.Context, logger logr.Logger, server string) (*types.AuthConfig, error) {
+	logger.WithName("acr-auth-provider")
 	match := acrRegex.FindAllString(server, -1)
 	if len(match) != 1 {
+		logger.V(2).Info("Invalid ACR url. ", server, "should match", acrRegex)
 		return nil, fmt.Errorf("invalid ACR url: %q should match %v", server, acrRegex)
 	}
 
 	loginServer := match[0]
-	err := retry(ctx, a.logger, 3, func() error {
+	err := retry(ctx, logger, 3, func() error {
 		return a.servicePrincipalToken.EnsureFreshWithContext(ctx)
 	})
 	if err != nil {
+		logger.Error(err, "Failed to refresh AAD token.")
```

Comment:
We bubble this error up and send it to NR so I wasn't sure if it was worth duplicating in the build logs/picked up by vector.